source/common/stats/stats_impl.cc

```diff
@@ -152,38 +166,72 @@ TagProducerImpl::TagProducerImpl(const envoy::config::metrics::v2::StatsConfig&
   }
 }
 
-std::string TagProducerImpl::produceTags(const std::string& stat_name,
-                                         std::vector<Tag>& tags) const {
-  tags.insert(tags.end(), default_tags_.begin(), default_tags_.end());
+void TagProducerImpl::addExtractorsMatching(absl::string_view name) {
+  int num_found = 0;
+  for (const auto& desc : Config::TagNames::get().descriptorVec()) {
+    if (desc.name == name) {
+      addExtractor(Stats::TagExtractorImpl::createTagExtractor(desc.name, desc.regex));
+      ++num_found;
```

Comment:
It *is* expected to go above 1.  That's one of the anticipated changes I have in the wings; my theory (which I need to isolate a test for) is that it's faster to do two different regexes when there are two different ways to generate a tag.  I think some of the catastrophic backtracking was due to too many alternatives in one regex, but I haven't totally wrapped my arms around that yet as I was focused on trying to avoid regexes completely.

Reply:
I see.  Since this is a future feature, can we toss an assert in for now, and remove if the implementation changes just to keep this PR self-contained?